certora/specs/HashConsensus.spec

```diff
@@ -0,0 +1,814 @@
+// using AccountingOracle as AccountingOracleContract
+
+
+/**************************************************
+ *               Methods Declaration              *
+ **************************************************/
+methods {
+    submitConsensusReport(bytes32 report, uint256 refSlot, uint256 deadline) => NONDET
+    getLastProcessingRefSlot() returns (uint256) => NONDET
+    getConsensusVersion() returns (uint256) => NONDET
+}
+
+/**************************************************
+ *                CVL FUNCS & DEFS                *
+ **************************************************/
+function saneTimeConfig() {
+    env e0; calldataarg args0;
+
+    require e0.msg.value == 0;                       // view functions revert is you send eth
+    require e0.block.timestamp > 1672531200;         // 01.01.2023 00:00:00
+    require e0.block.timestamp < 2524608000;         // 01.01.2050 00:00:00
+
+    uint256 slotsPerEpoch; uint256 secondsPerSlot; uint256 genesisTime;
+    slotsPerEpoch, secondsPerSlot, genesisTime = getChainConfig(e0);
+
+    require slotsPerEpoch == 32;                    // simplification, must be required at constructor
+    require secondsPerSlot == 12;                   // simplification, must be required at constructor
+    require genesisTime < e0.block.timestamp;       // safe assumption, must be required at constructor
+
+    uint256 initialEpoch; uint256 epochsPerFrame; uint256 fastLaneLengthSlots;
+    initialEpoch, epochsPerFrame, fastLaneLengthSlots = getFrameConfig(e0);
+    require epochsPerFrame > 0;                     // constructor already ensures this
+    require epochsPerFrame < 31536000;              // assuming less than 1 year per frame
+
+    // assuming correct configuration of the frame, otherwise revert
+    require initialEpoch < (e0.block.timestamp - genesisTime) / (secondsPerSlot * slotsPerEpoch);
```

Comment:
Same here. I guess it relies on the constructor?
Might it be changed during the contract's lifecycle?